sign/sign.go

```diff
@@ -187,3 +210,26 @@ func (s *Signer) IsImageSigned(imageRef string) (bool, error) {
 
 	return s.impl.IsImageSignedInternal(ctx, imageRef)
 }
+
+// IdentityToken returns an identity token to perform keyless signing.
+// If there is one set in the options we will use that one. If not,
+// signer will try to get one from the cosign OIDC identity providers
+// if options.EnableTokenProviders is set
+func (s *Signer) IdentityToken() (string, error) {
+	tok := s.options.IdentityToken
+	if s.options.PrivateKeyPath == "" && s.options.IdentityToken == "" {
```

Comment:
We already check that `s.options.PrivateKeyPath == ""` before calling this method, is this additional check intentional?

Reply:
This one is intentional. For now, we don't have a dedicated function for signing keyless or with a key. I want to avoid the problems seen in the promoter where paths controlled by more than one flag behave in different ways. I added this here to make it consent if the function gets reused from somewhere else (like SignFile) 

At some point I think we should split signing keyless and with a key to two separate functions